server/selection_test.go

```diff
@@ -0,0 +1,250 @@
+		stake := big.NewInt(int64(1000 * i))
+
+		sessions[i] = &BroadcastSession{
+			OrchestratorInfo: &net.OrchestratorInfo{
+				TicketParams: &net.TicketParams{Recipient: addr.Bytes()},
+			},
+		}
+		stakes[i] = stake
+		stakeMap[addr] = stake
+	}
+
+	stakeRdr.SetStakes(stakeMap)
+	sel.Add(sessions)
+
+	// Run unknownSessListSelect() x100000 and record # of times a session weight is selected
+	// Each session has a unique stake weight so we will record the # of selections per stake weight
+	stakeCount := make(map[*big.Int]int)
+	for i := 0; i < 100000; i++ {
+		sess := sel.unknownSessListSelect()
+		addr := ethcommon.BytesToAddress(sess.OrchestratorInfo.TicketParams.Recipient)
+		stake := stakeMap[addr]
+		stakeCount[stake]++
+
+		// Call Add() to add the session back to unknownSessList
+		sel.Add([]*BroadcastSession{sess})
+	}
+
+	sort.Slice(stakes, func(i, j int) bool {
+		return stakes[i].Cmp(stakes[j]) < 0
+	})
+
+	// Check that higher stake weight sessions are selected more often than lower stake weight sessions
+	for i, stake := range stakes[:len(stakes)-1] {
+		nextStake := stakes[i+1]
+		assert.Less(t, stakeCount[stake], stakeCount[nextStake])
+	}
+}
+
+func TestMinLSSelector_UnknownSessListSelect_UniformWeights(t *testing.T) {
+	stakeRdr := newStubStakeReader()
+	sel := NewMinLSSelector(stakeRdr, 1.0)
+
+	sessions := make([]*BroadcastSession, 10)
+	stakeMap := make(map[ethcommon.Address]*big.Int)
+	for i := 0; i < 10; i++ {
+		addr := ethcommon.BytesToAddress([]byte(strconv.Itoa(i)))
+
+		sessions[i] = &BroadcastSession{
+			OrchestratorInfo: &net.OrchestratorInfo{
+				TicketParams: &net.TicketParams{Recipient: addr.Bytes()},
+			},
+		}
+		stakeMap[addr] = big.NewInt(1000)
+	}
+
+	stakeRdr.SetStakes(stakeMap)
+	sel.Add(sessions)
+
+	// Run unknownSessListSelect() x1000000 and record # of times a session is selected
+	sessCount := make(map[*BroadcastSession]int)
+	for i := 0; i < 1000000; i++ {
+		sess := sel.unknownSessListSelect()
+		sessCount[sess]++
+
+		// Call Add() to add the session back to unknownSessList
+		sel.Add([]*BroadcastSession{sess})
+	}
+
+	// Check that the difference between the selection count of each session is less than some small delta
+	maxDelta := .015
+	for i, sess := range sessions[:len(sessions)-1] {
+		nextSess := sessions[i+1]
+		diff := math.Abs(float64(sessCount[sess] - sessCount[nextSess]))
+		delta := diff / float64(sessCount[sess])
+		assert.Less(t, delta, maxDelta)
+	}
+}
+
+func TestMinLSSelector_UnknownSessListSelect_NilStakeReader(t *testing.T) {
+	sel := NewMinLSSelector(nil, 1.0)
+
+	sessions := make([]*BroadcastSession, 10)
+	for i := 0; i < 10; i++ {
+		sessions[i] = &BroadcastSession{}
+	}
+
+	sel.Add(sessions)
+
+	i := 0
+	// Check that we select sessions based on the order of unknownSessList and that the size of
+	// unknownSessList decreases with each selection
+	for sel.Size() > 0 {
+		sess := sel.unknownSessListSelect()
+		assert.Equal(t, sess, sessions[i])
```

Comment:
this will always be equal as we are going to compare values here and not the pointers. To compare the pointers we have to explicitly say so.  So _I think_  in the current state the test will always pass.

Reply:
Fixed in 703441e - using [assert.Same()[https://godoc.org/github.com/stretchr/testify/assert#Same]